services/document_service.go

```diff
@@ -1,18 +1,37 @@
 package services
 
 import (
-	"time"
-
 	"github.com/docentre/docentre/models"
 	"github.com/docentre/docentre/repositories"
 )
 
 func CreateDocument(authorID uint) (models.Document, error) {
-	doc := models.Document{AuthorID: authorID, LastEditDate: time.Now()}
-	result := repositories.DB.Create(&doc)
+	doc := models.Document{AuthorID: authorID}
+	result := repositories.DB.Omit("ApprovedDate, ApproverID, ViewerID").Create(&doc)
 	if result.Error != nil {
 		return models.Document{}, result.Error
 	}
 
 	return doc, nil
 }
+
+func UpdateDocument(documentID uint, authorID uint, title string, content string, appendix string, status string, approverID uint) (uint, error) {
```

Comment:
Instead of accepting a large number of parameters, consider encapsulating them within a `struct`. This approach helps prevent users from inadvertently swapping the order of parameters, especially when they share common types.

Reply:
I'm not pretty sure. But this is controller's responsibility to make the object type into the flatten primitive type.
Otherwise, I only need to pass the entire request body inside Service?

Reply:
Since the `service` is where our business logic resides, it has the authority to define the structure it prefers, which doesn't necessarily have to consist of flattened primitives. This structure is recognized by the service, making it resilient to changes in the actual request body. The controller's responsibility is to convert the request body into this structure.

Reply:
The structure can currently be identical to the request body. While the conversion may seem redundant, I believe it's necessary to provide independence.